scrunch/datasets.py

```diff
@@ -2412,14 +2409,22 @@ def fork(self, description=None, name=None, is_published=False,
             **kwargs
         )
 
+        owner = kwargs.get("owner")
         if preserve_owner:
             body['owner'] = self.resource.body.owner
+        elif owner:
+            body["owner"] = (
+                owner if owner.startswith("http") else get_project(owner).url
+            )
+
         # not returning a dataset
         payload = shoji_entity_wrapper(body)
-        _fork = self.resource.forks.create(payload).refresh()
-        # return a MutableDataset always
-        fork_ds = MutableDataset(_fork)  # Fork has same editor as current user
-        return fork_ds
+        try:
+            _fork = self.resource.forks.create(payload).refresh()
+        except TaskProgressTimeoutError as exc:
+            _fork = exc.entity.wait_progress(exc.response).refresh()
```

Comment:
If you get a timeout,  you should continue to raise, this is an invalid call with the exception. If it timed out it quite likely failed, and the code flow shouldn't continue bc you won't have a fork to carry on with.

Reply:
@jjdelc **question:** Do you refer here to catching the timeout inside the fork() method directly? Because it is common practice for users to catch the timeouts on forks with a [progress tracking](https://github.com/Crunch-io/pycrunch/blob/master/src/pycrunch/progress.py#L21) against a larger timeout. We simply do not want to have this inside the method here?

Reply:
What I mean is that if pycrunch raised a timeout, there's no recovery. Calling `.wait_progress` with the response of the timeout is not going to work, and it won't achieve anything (maybe some python error) but this line looks like it is attempting some fallback way to recover from the timeout and still obtain a `_fork` instance from it.

Reply:
Yep - got it now, @jjdelc! Make sense. @shaikh-ma I think if we would want control the timeout AND also return the fork when it completed, we would need to do this:

```
... < catch & wait >
entity = exc.entity.refresh()
fork_ds = self.get_dataset(entity.body.id)
```

I think this _should_ work?

Reply:
The line `_fork = exc.entity.wait_progress(exc.response).refresh()` worked as expected (without giving any error) when I tried out with a dataset. 

![Handling TimeOut error](https://github.com/user-attachments/assets/b51cbf11-b836-4460-bbf6-df4621a81f12)

Reply:
So this looks like it is continuing to wait after the client has timed out, say pycrunch's timeout is 2m and the operation is taking longer than that.

On the first exception the client will raise TimeOut, but the server may still carry on updating the progress. So that's what that 2nd attempt is doing, it's giving it another chance to continue. That 2nd one on its own may also raise Timeout in case it takes longer than, now, 4 minutes. And you'll be back at the same situation.

This is equivalent of saying `timeout = 2*default_timeout`. You've effectively waited 2 times. 

It's a way to get around longer tasks, but code wise it reads very strange as it is not clear what the intention of this 2nd attempt is. Maybe add some comments, or even better if you want to increase the timeout then best to configure a longer timeout all together